src/test/unit_tests/test_s3select.js

```diff
@@ -0,0 +1,201 @@
+            resolve();
+        });
+    });
+    assert(writable.emit('error', new Error("I am an error event in an async loop.")));
+}
+
+
+mocha.describe('s3select', function() {
+
+    mocha.it('should create ok', function() {
+        const select_args = {
+            query: "select sum(int(_2)) from stdin;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const s3select = new s3select_utils.S3SelectStream(select_args);
+        s3select.destroy();
+    });
+
+    mocha.it('select star', async function() {
+        const select_args = {
+            query: "select * from stdin;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, csv_small_str);
+        assert.strictEqual(output, csv_small_str);
+    });
+
+    mocha.it('select sum with where', async function() {
+        const select_args = {
+            query: "select sum(int(_2)) from stdin where int(_1) % 3 = 1;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, csv_small_str);
+        assert.strictEqual(output, "1554");
+    });
+
+
+    mocha.it('select with header name', async function() {
+        const select_args = {
+            query: "select c from stdin where int(a) < 5;",
+            input_format: "CSV",
+            input_serialization_format: {"FieldDelimiter": ",", "RecordDelimiter": "\n", "FileHeaderInfo": "USE"},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, csv_small_str);
+        assert.strictEqual(output, "339\n741\n53\n233\n55\n");
+    });
+
+    mocha.it('select json', async function() {
+        const select_args = {
+            query: "select _1.intf from S3Object[*] s;",
+            input_format: "JSON",
+            input_serialization_format: {},
+            records_header_buf: s3select_utils.S3SelectStream.records_message_headers
+        };
+        const output = await run_sql(select_args, json_str);
+        assert.strictEqual(output, "3\n123\n");
+    });
+
+    mocha.it('double pipe close', async function() {
+        const readable = stream.Readable.from(["this is an input string."]);
+        const collect = new CollectChunks();
+        const transform = new FailingTransform(collect);
+        stream_utils.pipeline([transform, collect], true /*res is a write stream, no need for resume*/);
```

Comment:
Is the comment about this line? I didn't understand it.

Reply:
Yes, it explains why param is true.

Reply:
Did you mean `reuse` (as in `reuse_last_stream`)?
It is written `resume`.

Reply:
It's somewhat confusing, the parameter name is indeed "reuse_last_stream", but when it's set to true, then the last stream is not *resumed*.